### ISSUE TITLE
Testing partnership data

### ISSUE THREAD

**Issue by zharpaz** [phax/as2-lib#35]
Hi,
Implementing AS2, we want to validate a partnership's data before the first actual send. Can the library help with such a test?
Thank you for your help,
  Ziv

**Comment by phax** [comment]
What part of the partnership data you want to test? Please give some example.
Something like "does the alias exist in my keystore" or so?

**Comment by zharpaz** [comment]
I want to check that the remote URL is correct. If it needs user & password, that they are correct. If possible, that the certificate is correct.

**Comment by phax** [comment]
That's a tough thing to check, because there is no "ping" available. A server may (at least theoretically) block any other request than AS2 requests... How would you do that?

**Comment by zharpaz** [comment]
I thought to send a HEAD Query (as described in the as2 restart profile). At least servers that support restart will respond to it. What do you think?

**Comment by zharpaz** [comment]
I see. And as2-lib, does it supports HTTP/S? for sending?

**Comment by phax** [comment]
Yes, it does support http and https for sending of course. You can configure the certificate used, and you can even use a client certificate for sending. Also a proxy can be used
